src/formatDate.js

```diff
@@ -9,6 +9,58 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const splittedDate = date.split(fromFormat[3]);
+  const oldDateObj = {};
+  const result = [];
+
+  for (let i = 0; i < splittedDate.length; i++) {
+    oldDateObj[fromFormat[i]] = splittedDate[i];
+    // make new object via input date;
+  }
```

Comment:
This loop can be replaced with a more functional approach using `forEach` or `reduce` for better readability and less imperative code style.